packages/apalis-core/src/worker/ready.rs

```diff
@@ -83,30 +88,34 @@ where
                 }
             });
         }
-        while let Some(res) = futures::select! {
+        while let Some(res) = futures::select_biased! {
+            _ = ctx.shutdown.clone().fuse() => None,
             res = stream.next().fuse() => res,
-            _ = ctx.shutdown.clone().fuse() => None
         } {
+            let res: Result<Vec<_>, _> = res.into_iter().collect();
             let send_clone = send.clone();
             match res {
-                Ok(Some(item)) => {
-                    let svc = service
-                        .ready()
-                        .await
-                        .map_err(|e| WorkerError::ServiceError(format!("{e:?}")))?;
-                    let fut = svc.call(item);
-                    ctx.spawn(async move {
-                        fut.await;
-                    });
+                Ok(items) => {
+                    let mut futures = Vec::with_capacity(items.len());
+                    let items = items.into_iter().flatten();
+                    for item in items {
+                        let svc = service
+                            .ready()
```

Comment:
In an ideal situation we should rely on `ready` to fetch the next jobs. This can be worked on in v0.5

Reply:
Yeah sure This can be improved. 
What is your expected Release date for the 0.5 ? 
I can try to make another PR which improve the job fetching

Reply:
#126 Will solve the issue you are addressing.